### PR TITLE
record paused time and subtract from total time

### PR DESCRIPTION
**Pull request by aastein**
Record the time taken by the limiter to wait as `paused` time. Subtract the `paused` time from the total time since the last reported time.

### CODE REVIEW

**Comment by rgarcia** [comment]
Curious for the bigger context here--is there a worker where the 1 per second limiter gets hit often, making active percent less accurate? If so we can probably relax this limit

**Comment by aastein** [comment]
Most oneroser-csv-uploader jobs take less than 0.7 seconds, so the activePercent never reaches 70% causing the worker count to scale to 1 which then causes jobs to backup.

File: cmd/sfncli/sfncli.go [hunk 1]
```diff
@@ -126,9 +126,12 @@ func main() {
 			log.Info("getactivitytask-stop")
 		default:
 			cw.SetActiveState(false)
+			cw.SetPaused(true)
```
Comment (natebrennand) [inline cmd/sfncli/sfncli.go]: can we add a comment explaining the general context of not counting paused time towards inactive time?

**Comment by natebrennand** [comment]
@rgarcia: other context is that the behavior of how "active" time was tabulated changed in this PR - https://github.com/Clever/sfncli/pull/32
We thought not counting the paused time would be a better root cause fix than shifting where we call `SetActiveState`

File: cmd/sfncli/sfncli.go [hunk 2]
```diff
@@ -126,9 +126,14 @@ func main() {
 			log.Info("getactivitytask-stop")
 		default:
 			cw.SetActiveState(false)
+			// setting paused here so the time spent waiting for the limiter is not counted as time
```
Comment (mohit) [inline cmd/sfncli/sfncli.go]: Add a positive and a negative test for this new behavior.

File: cmd/sfncli/sfncli.go [hunk 3]
```diff
@@ -126,9 +126,14 @@ func main() {
...
+			// the task is inactive in the activePercent calcualtion
+			cw.SetPaused(true)
 			if err := limiter.Wait(mainCtx); err != nil {
+				cw.SetPaused(false)
```
Comment (mohit) [inline cmd/sfncli/sfncli.go]: Add comment why this is a place where we _must_ Unpause

File: cmd/sfncli/sfncli.go [hunk 4]
```diff
@@ -126,9 +126,14 @@ func main() {
...
 				continue
 			}
+			cw.SetPaused(false)
```
Comment (mohit) [inline cmd/sfncli/sfncli.go]: Same as above - add comment

**Pull request merged**